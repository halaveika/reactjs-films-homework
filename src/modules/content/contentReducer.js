export const initialState = {
  page: 0,
  results: [],
  total_pages: 0,
  total_results: 0,
  genres_array: [],
  video_url: '',
};

export const contentReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ADD_ITEMS':
      return {
        ...state,
        results: action.payload.results,
        page: action.payload.page,
        total_pages: action.payload.total_pages,
        total_results: action.payload.total_results,
      };
    case 'ADD_GENRES':
      return { ...state, genres_array: action.payload.genres };
    case 'ADD_VIDEO':
      return { ...state, video_url: action.payload };
    default:
      return state;
  }
};
