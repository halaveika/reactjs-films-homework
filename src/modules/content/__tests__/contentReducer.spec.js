import { contentReducer, initialState } from '../contentReducer';

describe('content reducer', () => {
  it('should return the initial state', () => {
    expect(contentReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle ADD_ITEMS', () => {
    const action = {
      type: 'ADD_ITEMS',
      payload: {
        results: [],
        page: 1,
        total_pages: 12,
        total_results: 55,
      },
    };
    expect(
      contentReducer(initialState,
        action),
    ).toMatchSnapshot();
  });

  it('should handle ADD_GENRES', () => {
    const action = {
      type: 'ADD_GENRES',
      payload: {
        genres: [213, 333, 45],
      },
    };
    expect(
      contentReducer(initialState,
        action),
    ).toMatchSnapshot();
  });

  it('should handle ADD_VIDEO', () => {
    const action = { type: 'ADD_VIDEO', payload: 'test_url' };
    expect(
      contentReducer(initialState,
        action),
    ).toMatchSnapshot();
  });
});
