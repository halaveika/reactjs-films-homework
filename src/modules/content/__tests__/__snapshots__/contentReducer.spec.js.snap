// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content reducer should handle ADD_DETAILS 1`] = `
Object {
  "details": Object {},
  "filter": "TRENDING",
  "genres_array": Array [],
  "isLoading": true,
  "page": 1,
  "pageSize": 20,
  "results": Array [],
  "searchValue": "",
  "total_pages": 0,
  "total_results": 0,
}
`;

exports[`content reducer should handle ADD_GENRES 1`] = `
Object {
  "details": Object {},
  "filter": "TRENDING",
  "genres_array": Array [
    Object {
      "id": 13,
      "name": "Adventure",
    },
    Object {
      "id": 18,
      "name": "Drama",
    },
    Object {
      "id": 20,
      "name": "Family",
    },
    Object {
      "id": 21,
      "name": "Fantasy",
    },
  ],
  "isLoading": true,
  "page": 1,
  "pageSize": 20,
  "results": Array [],
  "searchValue": "",
  "total_pages": 0,
  "total_results": 0,
}
`;

exports[`content reducer should handle ADD_ITEMS 1`] = `
Object {
  "details": Object {},
  "filter": "TRENDING",
  "genres_array": Array [],
  "isLoading": false,
  "page": 1,
  "pageSize": 20,
  "results": Array [],
  "searchValue": "",
  "total_pages": 12,
  "total_results": 55,
}
`;

exports[`content reducer should handle SET_CURRENT_PAGE 1`] = `
Object {
  "details": Object {},
  "filter": "TRENDING",
  "genres_array": Array [],
  "isLoading": true,
  "page": undefined,
  "pageSize": undefined,
  "results": Array [],
  "searchValue": "",
  "total_pages": 0,
  "total_results": 0,
}
`;

exports[`content reducer should handle SET_FILTER 1`] = `
Object {
  "details": Object {},
  "filter": "COMINGSOON",
  "genres_array": Array [],
  "isLoading": true,
  "page": 1,
  "pageSize": 20,
  "results": Array [],
  "searchValue": "",
  "total_pages": 0,
  "total_results": 0,
}
`;

exports[`content reducer should handle SET_LOADING 1`] = `
Object {
  "details": Object {},
  "filter": "TRENDING",
  "genres_array": Array [],
  "isLoading": false,
  "page": 1,
  "pageSize": 20,
  "results": Array [],
  "searchValue": "",
  "total_pages": 0,
  "total_results": 0,
}
`;

exports[`content reducer should handle SET_SEARCH_VALUE 1`] = `
Object {
  "details": Object {},
  "filter": "TRENDING",
  "genres_array": Array [],
  "isLoading": true,
  "page": 1,
  "pageSize": 20,
  "results": Array [],
  "searchValue": "test",
  "total_pages": 0,
  "total_results": 0,
}
`;

exports[`content reducer should return the initial state 1`] = `
Object {
  "details": Object {},
  "filter": "TRENDING",
  "genres_array": Array [],
  "isLoading": true,
  "page": 1,
  "pageSize": 20,
  "results": Array [],
  "searchValue": "",
  "total_pages": 0,
  "total_results": 0,
}
`;
